Source/WebCore/page/Page.cpp

```diff
@@ -1810,6 +1810,10 @@ void Page::updateRendering()
         }
     });
 
+    runProcessingStep(RenderingUpdateStep::UpdateValidationMessagePositions, [] (Document& document) {
+        document.adjustValidationMessagePositions();
+    });
```

Comment:
Since this relies on up-to-date layout information, I think it might be more appropriate to do in `Page::doAfterUpdateRendering()` (note the first comment in that function)

Reply:
Ah wait no, `adjustBubblePosition()` could trigger layout by calling `m_bubble->setInlineStyleProperty`, so this is probably good...

Reply:
Right. FWIW, line 1734 above updates layout before running this code.